test/integration/src/test/java/io/pravega/test/integration/LargeEventTest.java

```diff
@@ -0,0 +1,523 @@
+        createScopeStream(SCOPE_NAME, streamName, config);
+        generateWriteEventData(NUM_WRITERS, Serializer.MAX_EVENT_SIZE * 5);
+
+        try (ConnectionFactory connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder().build());
+             ClientFactoryImpl clientFactory = new ClientFactoryImpl(SCOPE_NAME, controller, connectionFactory);
+             ReaderGroupManager readerGroupManager = new ReaderGroupManagerImpl(SCOPE_NAME, controller, clientFactory)) {
+            // Start writing events to the stream.
+            val writers = createEventWriters(streamName, NUM_WRITERS, clientFactory);
+            // Create a ReaderGroup.
+            createReaderGroup(readerGroupName, readerGroupManager, streamName);
+            // Create Readers.
+            val readers = createEventReaders(NUM_READERS, clientFactory, readerGroupName);
+            Futures.allOf(writers).get();
+            ExecutorServiceHelpers.shutdown(writerPool);
+            stopReadFlag.set(true);
+            Futures.allOf(readers).get();
+            ExecutorServiceHelpers.shutdown(readerPool);
+
+            log.info("Deleting ReaderGroup: {}", readerGroupName);
+            readerGroupManager.deleteReaderGroup(readerGroupName);
+        }
+
+        validateCleanUp(streamName);
+        validateEventReads(eventsReadFromPravega, eventsWrittenToPravega);
+    }
+
+    @Test
+    public void testReadWriteWithSegmentStoreRestart() throws ExecutionException, InterruptedException {
+    String readerGroupName = "testLargeEventFailoverReaderGroup";
+        String streamName = "SegmentStoreRestart";
+        StreamConfiguration config = getStreamConfiguraton(NUM_READERS);
+        createScopeStream(SCOPE_NAME, streamName, config);
+        generateWriteEventData(NUM_WRITERS, Serializer.MAX_EVENT_SIZE * 5);
+
+        try (ConnectionFactory connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder().build());
+             ClientFactoryImpl clientFactory = new ClientFactoryImpl(SCOPE_NAME, controller, connectionFactory);
+             ReaderGroupManager readerGroupManager = new ReaderGroupManagerImpl(SCOPE_NAME, controller, clientFactory)) {
+            // Start writing events to the stream.
+            val writers = createEventWriters(streamName, NUM_WRITERS, clientFactory);
+            // Create a ReaderGroup.
+            createReaderGroup(readerGroupName, readerGroupManager, streamName);
+            // Create Readers.
+            val readers = createEventReaders(NUM_READERS, clientFactory, readerGroupName);
+            Futures.allOf(writers).get();
+            ExecutorServiceHelpers.shutdown(writerPool);
+            stopReadFlag.set(true);
+
+            // Reset the server, in effect clearing the AppendProcessor and PravegaRequestProcessor.
+            this.server = new PravegaConnectionListener(false, servicePort, store, tableStore, serviceBuilder.getLowPriorityExecutor());
```

Comment:
Please close the existing server instance, otherwise it might not clean up any allocated resources.